testing/test_unittest.py

```diff
@@ -208,10 +208,14 @@ def test_demo(self):
 
     """
     )
+
     pytester.inline_run("-s", testpath)
     gc.collect()
+
+    # Either already destroyed, or didn't run setUp.
     for obj in gc.get_objects():
-        assert type(obj).__name__ != "TestCaseObjectsShouldBeCleanedUp"
+        if type(obj).__name__ == "TestCaseObjectsShouldBeCleanedUp":
+            assert not hasattr(obj, "an_expensive_obj")
```

Comment:
Where does this `an_expensive_obj` comes from? Perhaps leaving a comment explaining it.

Reply:
I think it's maybe clearer when reading the entire test?